docs/docs/developers/docs/concepts/accounts/keys.md

```diff
@@ -1,155 +1,307 @@
+Since signing keys are application-defined, how the public key is stored is entirely up to the account contract design. Each approach has different trade-offs:
+
+### Immutable Private Note (Recommended for Most Cases)
+
+Stores the key in a private note that is never nullified, providing the best balance of security and cost.
 
 #include_code public_key noir-projects/noir-contracts/contracts/account/schnorr_account_contract/src/main.nr rust
 
-:::note
-When it comes to storing the signing key in a private note, there are several details that rely on the wallets:
+**Pros:**
+
+- No additional cost per transaction
+- Key remains private
+- Simple implementation
 
-- A note with a key is managed similar to any other private note. Wallets are expected to backup all the notes so that they can be restored on another device (e.g. if the user wants to move to another device).
-- The note with the key might exist locally only (in PXE) or it can be broadcasted as an encrypted note by the wallet to itself. In the second case, this note will also exist on Aztec.
-  :::
+**Cons:**
 
-#### Using Delayed Public Mutable state
+- Cannot rotate the key
+- If key is compromised, must deploy a new account
+
+:::note Wallet Responsibilities
+Private note storage relies on wallet infrastructure:
+
+- Wallets must backup all notes for device restoration
+- Notes can exist locally (PXE only) or be broadcast encrypted to yourself (exists onchain)
+:::
 
-By [Delayed Public Mutable](../../guides/smart_contracts/how_to_define_storage.md#delayed-public-mutable) we mean privately readable publicly mutable state.
+### Mutable Private Note
 
-To make public state accessible privately, there is a delay window in public state updates. One needs this window to be able to generate proofs client-side. This approach would not generate additional nullifiers and commitments for each transaction while allowing the user to rotate their key. However, this causes every transaction to now have a time-to-live determined by the frequency of the delayed mutable state, as well as imposing restrictions on how fast keys can be rotated due to minimum delays.
+Stores the key in a note that can be updated, enabling key rotation.
 
-#### Reusing some of the in-protocol keys
+**Pros:**
 
-It is possible to use some of the key pairs defined in protocol (e.g. incoming viewing keys) as the signing key. Since this key is part of the address preimage, it can be validated against the account contract address rather than having to store it. However, this approach is not recommended since it reduces the security of the user's account.
+- Supports key rotation
+- Key remains private
+
+**Cons:**
+
+- Each transaction nullifies and recreates the note (expensive)
+- Higher cost per transaction
+
+### Delayed Public Mutable State
```

Comment:
I know this is possible, but do we want to recommend it as an option? what is a case where having the public key be public be useful?